spec/db/seeder_spec.rb

```diff
@@ -0,0 +1,202 @@
+      let(:entries) { [{name: "sha1000"}, {name: "sha9001"}]}
+      let(:type) { :fixity_algs }
+      let(:fabricator_name) { :fixity_alg }
+      let(:display_field) { :name }
+    end
+  end
+
+  describe "create protocols" do
+    it_behaves_like "a seedable class", Protocol do
+      let(:entries) { [{name: "p1"}, {name: "p2"}]}
+      let(:type) { :protocols }
+      let(:fabricator_name) { :protocol }
+      let(:display_field) { :name }
+    end
+  end
+
+  describe "create storage_regions" do
+    it_behaves_like "a seedable class", StorageRegion do
+      let(:entries) { [{name: "r1"}, {name: "r2"}]}
+      let(:type) { :storage_regions }
+      let(:fabricator_name) { :storage_region }
+      let(:display_field) { :name }
+    end
+  end
+
+  describe "create storage_types" do
+    it_behaves_like "a seedable class", StorageType do
+      let(:entries) { [{name: "t1"}, {name: "t2"}]}
+      let(:type) { :storage_types }
+      let(:fabricator_name) { :storage_type }
+      let(:display_field) { :name }
+    end
+  end
+
+  describe "create members" do
+    it_behaves_like "a seedable class", Member do
+      let(:entries) do
+        2.times.map do
+          { member_id: SecureRandom.uuid, name: Faker::Company.name, email: Faker::Internet.email }
+        end
+      end
+      let(:type) { :members }
+      let(:fabricator_name) { :member }
+      let(:display_field) { :name }
+    end
+  end
+
+  describe "create nodes" do
+    let(:entry) do
+      {
+        namespace: Faker::Lorem.word,
+        name: Faker::Company.name,
+        storage_region: Fabricate(:storage_region).name,
+        storage_type: Fabricate(:storage_type).name,
+        fixity_algs: [Fabricate(:fixity_alg).name],
+        protocols: [Fabricate(:protocol).name],
+        api_root: Faker::Internet.url,
+        private_auth_token: Faker::Crypto.sha256,
+        auth_credential: Faker::Crypto.sha256
+      }
+    end
+
+    let(:skip_message) { "Skipped existing Node: #{entry[:namespace]}" }
+
+    it "creates Node records" do
+      seeder.create(:nodes, [entry])
+      created_node = Node.find_by_namespace(entry[:namespace])
+      expect(created_node.name).to eql(entry[:name])
+      expect(created_node.storage_region.name).to eql(entry[:storage_region])
+      expect(created_node.storage_type.name).to eql(entry[:storage_type])
+      expect(created_node.fixity_algs.pluck(:name)).to contain_exactly(*entry[:fixity_algs])
+      expect(created_node.protocols.pluck(:name)).to contain_exactly(*entry[:protocols])
+      expect(created_node.api_root).to eql(entry[:api_root])
+      expect(created_node.private_auth_token)
+        .to eql(Node.send(:generate_hash, entry[:private_auth_token]))
+      expect(created_node.auth_credential).to eql(entry[:auth_credential])
+    end
+    it "skips existing Node records" do
+      Fabricate(:node, namespace: entry[:namespace])
+      seeder.create(:nodes, [entry])
+      expect(logger.info_messages).to eql([skip_message])
```

Comment:
On the fence about this.  The script is meant to be idempotent--e.g. a new member gets addded, run the entire thing.  So, skips will be normal.  I didn't want to issue a warning about normal behavior.